### PR TITLE
Move goffice to library and update to 0.10.35

### PR DESCRIPTION
**Pull request by ghost**
Added PR

### CODE REVIEW

File: components/library/goffice/Makefile [hunk 1]
```diff
@@ -10,20 +10,21 @@
...
 #
 
-include ../../make-rules/shared-macros.mk
+include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME=		goffice
-COMPONENT_VERSION=	0.8.17
-COMPONENT_REVISION=	1
+COMPONENT_VERSION=	0.10.35
+COMPONENT_REVISION= 1
```
Comment (pyhalov) [inline components/library/goffice/Makefile]: Please, remove COMPONENT_REVISION

File: components/library/goffice/Makefile [hunk 2]
```diff
@@ -49,12 +50,29 @@ CONFIGURE_ENV+= PYTHON=$(PYTHON)
...
+build:		$(BUILD_32_and_64)
 
-install:	$(INSTALL_32)
+install:	$(INSTALL_32_and_64)
 
-test:		$(NO_TESTS)
+# To prevent "libtool_install_magic: unbound variable"
+unexport SHELLOPTS
+
+#test:		$(NO_TESTS)
```
Comment (pyhalov) [inline components/library/goffice/Makefile]: remove this unnecessary comment

File: components/library/goffice/Makefile [hunk 3]
```diff
@@ -54,7 +54,6 @@ build:		$(BUILD_32_and_64)
 
 install:	$(INSTALL_32_and_64)
 
-# To prevent "libtool_install_magic: unbound variable"
 unexport SHELLOPTS
```
Comment (pyhalov) [inline components/library/goffice/Makefile]: I meant another comment - this one was useful, #test: $(NO_TESTS) - is not

File: components/library/goffice/Makefile [hunk 4]
```diff
@@ -49,12 +50,27 @@ CONFIGURE_ENV+= PYTHON=$(PYTHON)
...
 
-install:	$(INSTALL_32)
+install:	$(INSTALL_32_and_64)
 
-test:		$(NO_TESTS)
+unexport SHELLOPTS
+
+test:		$(TEST_32_and_64)
 
 BUILD_PKG_DEPENDENCIES = $(BUILD_TOOLS)
```
Comment (pyhalov) [inline components/library/goffice/Makefile]: Haven't spotted this - remove BUILD_PKG_DEPENDENCIES and include $(WS_TOP)/make-rules/depend.mk lines - this is legacy , we don't use them.

Comment (pyhalov) [reply components/library/goffice/Makefile]: Also change $(WS_TOP)/make-rules to $(WS_MAKE_RULES)

**Comment by pyhalov** [comment]
please, remove COMPONENT_REVISION when update COMPONENT_VERSION

File: components/library/goffice/Makefile [hunk 5]
```diff
@@ -30,9 +30,9 @@ COMPONENT_LICENSE=	GPLv2
 COMPONENT_LICENSE_FILE=	COPYING
 COMPONENT_FMRI=		library/desktop/goffice
 
-include $(WS_TOP)/make-rules/prep.mk
-include $(WS_TOP)/make-rules/configure.mk
-include $(WS_TOP)/make-rules/ips.mk
+include $(WS_MAKE_RULES)/make-rules/prep.mk
```
Comment (pyhalov) [inline components/library/goffice/Makefile]: this should have become $(WS_MAKE_RULES)/prep.mk and so on

File: components/library/goffice/Makefile [hunk 6]
```diff
@@ -10,28 +10,28 @@
...
+include $(WS_MAKE_RULES)/make-rules/configure.mk
+include $(WS_MAKE_RULES)/make-rules/ips.mk
 
 PATH=/usr/gnu/bin:/usr/bin
```
Comment (pyhalov) [inline components/library/goffice/Makefile]: Please, replace it with PATH=$(PATH.gnu)

File: components/library/goffice/patches/sun_go-math.patch [hunk 7]
```diff
@@ -0,0 +1,16 @@
+--- goffice-0.10.35/goffice/math/go-math.h.sun	Sat Apr 11 22:14:13 2015
++++ goffice-0.10.35/goffice/math/go-math.h	Fri Aug 25 14:54:57 2017
+@@ -2,7 +2,12 @@
+ #define __GO_MATH_H
+ 
+ #include <goffice/goffice-features.h>
+-#include <math.h>
++//solaris hides finite in ieeefp.h, not math.h
```
Comment (pyhalov) [inline components/library/goffice/patches/sun_go-math.patch]: If it's supposed to be c89-compatible, perhaps we should change // to /**/  (as we ship this header, we  can't  do assumptions about its consumers).

**Comment by ghost** [comment]
Some of the changes we have made in the last few days in the PR's confuse me.
Why did it allow me to build the packages anyway?
I have all three packages running now.

**Comment by pyhalov** [comment]
What exact changes do confuse you?

**Comment by ghost** [comment]
include $(WS_MAKE_RULES)/make-rules/prep.mk changed to
include $(WS_MAKE_RULES)/prep.mk
It looks like one would not work because it looks like a different path.

**Comment by ghost** [comment]
Also, is there a good doc on solaris verus gnu on things like where is finite.h, no pty.h, no util.h and such.

**Comment by pyhalov** [comment]
$(WS_MAKE_RULES)/make-rules/prep.mk did not work.
There's Red Hat Linux to Oracle Solaris porting guide, but it is absolutely not complete.

**Pull request merged**